packages/react-charting/UnitTests/AreaChartUT.test.tsx

```diff
@@ -0,0 +1,683 @@
+      const legend2 = screen.queryByText('legend2');
+      expect(legend2).toBeDefined();
+      fireEvent.click(legend2!);
+      // Assert
+      const areas = getById(container, /line-areaChart/i);
+      expect(areas.length).toBe(3);
+      expect(areas[1].getAttribute('opacity')).toEqual('0');
+    },
+  );
+});
+
+runTest('_updateCircleFillColor', () => {
+  testWithoutWait(
+    'Should return the line color if neither the nearest circle nor the active point is highlighted',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      // Assert
+      points.forEach(point => {
+        expect(point.getAttribute('fill')).toEqual('green');
+      });
+    },
+  );
+
+  testWithoutWait(
+    'Should return white if the nearest circle is highlighted and the circle is not clicked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[6]);
+      expect(points[6].getAttribute('fill')).toEqual('#ffffff');
+    },
+  );
+
+  testWithoutWait(
+    'Should return the line color if the nearest circle is highlighted and the circle is clicked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[0]);
+      fireEvent.click(points[0]);
+      expect(points[0].getAttribute('fill')).toEqual('green');
+    },
+  );
+
+  testWithoutWait(
+    'Should return white if the active point is highlighted and the circle is not clicked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.focus(points[0]);
+      expect(points[0].getAttribute('fill')).toEqual('#ffffff');
+    },
+  );
+
+  testWithoutWait(
+    'Should return the line color if the active point is highlighted but the circle is clicked',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[0]);
+      fireEvent.click(points[0]);
+      fireEvent.mouseOver(points[1]);
+      fireEvent.click(points[1]);
+      expect(points[1].getAttribute('fill')).toEqual('green');
+    },
+  );
+});
+
+runTest('_getCircleRadius', () => {
+  testWithoutWait(
+    'Should return 1 if isCircleClicked is true and nearestCircleToHighlight matches xDataPoint',
+    AreaChart,
+    { data: chartData },
+    container => {
+      const points = getById(container, /circle/i);
+      expect(points).toHaveLength(10);
+      fireEvent.mouseOver(points[1]);
+      fireEvent.click(points[0]);
+      // nearest circle to highlight has index 6
+      expect(points[6].getAttribute('r')).toEqual('1');
```

Comment:
what is the difference between this and the next test

Reply:
In this test, activePoint does not matches the point circle whose radius is checked and in the next test activePoint matches the point circle

Reply:
mention this explanation as a comment in the test case